hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonRecordReader.java

```diff
@@ -80,7 +80,7 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext context)
     }
     // It should use the exists tableBlockInfos if tableBlockInfos of queryModel is not empty
     // otherwise the prune is no use before this method
-    if (!queryModel.isFG()) {
+    if (queryModel.getTableBlockInfos().isEmpty()) {
```

Comment:
If use  (queryModel.getTableBlockInfos().isEmpty()),  when the prune result is empty by FG in search mode, it will use the original TableBlockInfos and execute again, it mean the FG no use in this scenario. So we cannot change it to queryModel.getTableBlockInfos().isEmpty()

Reply:
I have added a new RecordReader called `IndexedRecordReader`, I am using this one in search mode now. So this problem will not come. I will remove line 83.